src/ipc-handlers.ts

```diff
@@ -571,20 +570,16 @@ export async function saveOnboarding(
 
 export async function executeWPCLiInline(
 	_event: IpcMainInvokeEvent,
-	{ projectPath, args }: { projectPath: string; args: string }
-) {
-	const wpCliArgs = parse( args );
-
-	// The parsing of arguments can include shell operators like `>` or `||` that the app don't support.
-	const isValidCommand = wpCliArgs.every(
-		( arg: unknown ) => typeof arg === 'string' || arg instanceof String
-	);
-	if ( ! isValidCommand ) {
-		throw Error( `Can't execute wp-cli command with arguments: ${ args }` );
+	{ siteId, args }: { siteId: string; args: string }
+): Promise< WpCliResult > {
+	if ( SiteServer.isDeleted( siteId ) ) {
```

Comment:
Why do we need both "deleted" sites and "not found" sites? Why are these two states not equal? I.e., when deleting a site, would removing the site entity not also satisfy a "deleted" site state?

Reply:
I added both conditions for covering two different scenarios:
- **Deleted site:** This is an expected condition that we want to handle gracefully, as it might happen that the app runs a  `wp-cli` command on a deleted site (as described in the PR's description).
- **Not-defined site:** This would be an unexpected case where we'll throw an exception. This is basically a safeguard against a call with wrong arguments as we have on othershandlers, not to cover a real case.